byron/ledger/executable-spec/src/Ledger/Core.hs

```diff
@@ -159,18 +161,18 @@ class Relation m where
 
   -- | Domain restriction
   --
-  (◁), (◃), (<|) :: Ord (Domain m) => Set (Domain m) -> m -> m
+  (◁), (◃), (<|) :: (Ord (Domain m), Ord (Range m)) => Set (Domain m) -> m -> m
   s ◃ r = s ◁ r
   s <| r = s ◁ r
 
   -- | Domain exclusion
   --
-  (⋪), (</|) :: Ord (Domain m) => Set (Domain m) -> m -> m
+  (⋪), (</|) :: (Ord (Domain m), Ord (Range m)) => Set (Domain m) -> m -> m
   s </| r = s ⋪ r
 
   -- | Range restriction
   --
-  (▹), (|>) :: Ord (Range m) => m -> Set (Range m) -> m
+  (▹), (|>) :: (Ord (Domain m), Ord (Range m)) => m -> Set (Range m) -> m
```

Comment:
Hmmm, it looks a bit strange that we need to add these constraints just because we modified one of the instances of this class. Besides the new `Bimap` instance, is there a reason why we need these new constraints?

Reply:
It's because of this :(

`Bimap.filter :: (Ord a, Ord b) => (a -> b -> Bool) -> Bimap a b -> Bimap a b`

Should I perhaps find another route such that we don't need to make this change?

Reply:
I'd say that these constraints belong to the instance, so for instance, if you'd define:

```haskell
  r ▹ s = Bimap.filter (\_ v -> Set.member v s) r
```

you'd need the `Ord k` constraint, which you can add in the instance declaration:

```haskell
instance Ord k => Relation (Bimap k v) where
```

and the class declaration remains unaffected. 

On the other hand, my guess is that the existing constraint `Ord (Range m)` is there because we use the type `Set (Range m)` (although I guess we should be able to define the class without it, but then all the instances wold have to define the `Ord` constraint, which is kind of annoying, since we already know that we'll be using a set...)

But ping @ruhatch or @nc6 to be sure :)

Reply:
Thanks! I think I've addressed this concern now.

Reply:
Great!